source/common/tls/utility.cc

```diff
@@ -226,6 +226,62 @@ std::string Utility::generalNameAsString(const GENERAL_NAME* general_name) {
     }
     break;
   }
+  case GEN_OTHERNAME: {
+    ASN1_TYPE* value = general_name->d.otherName->value;
+    if (value == nullptr) {
+      break;
+    }
+    switch (value->type) {
```

Comment:
Should the matcher specify the expected type to avoid any potential bypass of the checks, similar to the change made in #18628? Or is that overkill here? cc @yanavlasov @pradeepcrao .

Reply:
@yanavlasov  @pradeepcrao friendly ping for any thoughts here. There is admittedly room for bypass, but I'm not sure if it the principle to create a distinction DNS/IP/EMAIL/URI applies here between primitive data types. Happy to make any changes if required.

Reply:
As a follow up, I checked the RFC for this. Nothing is clearly stated about the OtherName semantics in the SAN [subsection](https://datatracker.ietf.org/doc/html/rfc5280#section-4.2.1.6) - pretty much that the OID speaks for the type and the value speaks for the name. The name constraints [subsection](https://datatracker.ietf.org/doc/html/rfc5280#section-4.2.1.6) and the above section however go into detail as to what is expected of a DNS, URI, IP, EMAIL SAN and the format and semantics it should take and when certificates should be rejected. Adding #18628 helps us conform to that. Having an EMAIL SAN with "google.com" matched against a DNS expectation would break it since DNS SAN would be expected to have certain semantics. For othername SAN, they just specify this: 

>  The syntax and semantics for name constraints for otherName, ediPartyName, and registeredID are not defined by this specification, however, syntax and semantics for name constraints for other name forms may be specified in other documents.

Considering we have an OID matching which is probably what should define type here, it might just be enough based on my understanding of RFC.

On a side note, I also noticed another RFC which tries to define [semantics](https://datatracker.ietf.org/doc/html/rfc4985) very specific to an usecase, but nothing generic.